moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp

```diff
@@ -1312,7 +1313,23 @@ void TrajectoryExecutionManager::executeThread(const ExecutionCompleteCallback&
 
   // only report that execution finished successfully when the robot actually stopped moving
   if (last_execution_status_ == moveit_controller_manager::ExecutionStatus::SUCCEEDED)
-    waitForRobotToStop(*trajectories_[i - 1]);
+  {
+    execution_trajectories_mutex_.lock();
```

Comment:
Use existing `execution_state_mutex_` instead:
```suggestion
    std::scoped_lock slock(execution_state_mutex_);
```